File: src/app/model/entity/person-title.ts
import { Person } from './person';
import { Title } from './title';
import { DataStatus } from '../../core/model/data-status';
import { DataAction } from '../../core/model/data-action';

export class PersonTitle {

  person: Person;
  title: Title;

  status: DataStatus;
  createdBy: string;
  updatedBy: string;
  lastUpdated: Date;
  selected: boolean;
  action: DataAction;
  isDataChanged: boolean;

}

File: src/main/java/com/apschulewitz/resdb/refdata/model/entity/PersonTitle.java
package com.apschulewitz.resdb.refdata.model.entity;

import com.apschulewitz.resdb.common.model.VersionableDataEntity;
import com.apschulewitz.resdb.common.model.entity.DataOperation;
import com.apschulewitz.resdb.common.model.entity.Title;
import com.apschulewitz.resdb.common.model.entity.VersionStatus;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.Tolerate;

import javax.persistence.*;
import java.time.ZonedDateTime;

@Data
@Builder
@Entity
@Table(name = "resdb_person_title")
public class PersonTitle implements VersionableDataEntity<Long> {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  @ManyToOne
  @JoinColumn(name = "person_id")
  private Person person;

  @ManyToOne
  private Title title;

  @Column(nullable = false)
  private Integer position;

  @Column
  private VersionStatus status;

  @Column
  private String createdBy;

  @Column
  private String updatedBy;

  @Column(name = "version_no")
  private Long versionNumber;

  @Column(name = "last_updated")
  private ZonedDateTime lastUpdated;

  private transient DataOperation operation;
  @Tolerate
  public PersonTitle() {

  }
}
